core/src/main/java/org/apache/iceberg/rest/RESTCatalog.java

```diff
@@ -69,18 +75,66 @@
   private Object conf = null;
   private FileIO io = null;
 
+  public RESTCatalog() {
+    this(new HTTPClientFactory());
+  }
+
   RESTCatalog(Function<Map<String, String>, RESTClient> clientBuilder) {
     this.clientBuilder = clientBuilder;
   }
 
   @Override
   public void initialize(String name, Map<String, String> props) {
+    Map<String, String> mergedProps = fetchAndMergeWithServerConfig(props);
+    initializeInternal(name, mergedProps);
+  }
+
+  /**
+   * Performs the normal initialization steps after calling the config route and merging the initial
+   * configuration with any configuration provided by the server.
+   * <p>
+   * See {@link RESTCatalogConfigResponse} for details on merging the initial client and server configuration.
+   *
+   * @param name  Name of this RESTCatalog
```

Comment:
I think that arg names should match the variable names. Could you make the case match?

Reply:
Maybe this comment is dated, but if you're referring to the parameter names in the javadoc comment, they do match.

If you're referring to making the parameter named `properties`, I think that will cause a shadowed variable error in error prone since this isn't a constructor. Let me double check.

Reply:
Oh, looks like my bad. Sorry about that, not sure why I didn't see `name`.

Reply:
Ah no worries. It happens.